packages/worker/src/worker.ts

```diff
@@ -1297,30 +1302,33 @@ export class Worker {
    * This function does not throw, it will log in case of missing sinks
    * or failed sink function invocations.
    */
-  protected async processSinkCalls(externalCalls: SinkCall[], info: WorkflowInfo, isReplaying: boolean): Promise<void> {
+  protected async processSinkCalls(externalCalls: SinkCall[], workflowInfo: WorkflowInfo): Promise<void> {
+    // When processing workflows through runReplayHistories, Core may still send non-replay activations on the very
+    // last Workflow Task in some cases. Though Core is technically exact here, the fact that sinks marked with
+    // callDuringReplay = false may get called on a replay worker is definitely a surprising behavior. For that
+    // reason, we extend the isReplaying flag in this case to also include anything running under in a replay worker.
+    const isReplaying = workflowInfo.unsafe.isReplaying || this.isReplayWorker;
     const { sinks } = this.options;
-    await Promise.all(
```

Comment:
Not convinced we should be calling these sequentially, if a user has many sink calls that do IO this will delay the completion of WFTs.
We should also consider failing activations on failed sink calls, users may be relying on sinks for reporting.